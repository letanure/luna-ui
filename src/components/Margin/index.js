import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Margin component.
 * E.g.:
 * ```html
 * <Margin selector="elements" value="1px 3px 4px 2px" />
 * ```
 */
class MarginComponent extends React.PureComponent {
  render() {
    const { selector, value } = this.props;

    const Margin = styled.div`
      & ${selector} {
        margin: ${value};
      }
    `;

    return <Margin {...this.props}>{this.props.children}</Margin>;
  }
}

MarginComponent.propTypes = {
  children: PropTypes.node,
  /**
   * Selects the element to receive padding value
   */
  selector: PropTypes.string,
  /**
   * Value of padding to selected elements
   */
  value: PropTypes.string
};

MarginComponent.defaultProps = {
  children: null,
  selector: '*',
  value: '11px'
};

export default MarginComponent;
