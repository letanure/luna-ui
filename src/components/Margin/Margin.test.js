import React from 'react';
import renderer from 'react-test-renderer';

import Margin from './index.js';

describe('<Margin />', () => {
  const content =
    <h1>Snap Test Rocks</h1>
  ;

  const component =
    <Margin>{content}</Margin>;

  const componentSelector =
    <Margin selector="h1">{content}</Margin>;

  const componentValue =
    <Margin value="10px 20px 20px 30px">{content}</Margin>;

  const componentSelectAndValue =
    <Margin selector="h1" value="10px 20px 20px 30px">{content}</Margin>;

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Selector', () => {
    const tree = renderer
      .create(componentSelector)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Value', () => {
    const tree = renderer
      .create(componentValue)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Selector and Value', () => {
    const tree = renderer
      .create(componentSelectAndValue)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
