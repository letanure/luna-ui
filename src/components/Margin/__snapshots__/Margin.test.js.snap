// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Margin /> toMatchSnapshot 1`] = `
<div
  className="sc-bdVaJa hGGSnS"
  value="11px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;

exports[`<Margin /> toMatchSnapshot: Selector 1`] = `
<div
  className="sc-bwzfXH jxKUMT"
  value="11px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;

exports[`<Margin /> toMatchSnapshot: Selector and Value 1`] = `
<div
  className="sc-bxivhb gdnQvO"
  value="10px 20px 20px 30px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;

exports[`<Margin /> toMatchSnapshot: Value 1`] = `
<div
  className="sc-htpNat bxEBzI"
  value="10px 20px 20px 30px"
>
  <h1>
    Snap Test Rocks
  </h1>
</div>
`;
